### ISSUE TITLE
Add SigmoidF1 & UnboundedF1 Loss functions

### ISSUE THREAD

**Issue by MohamedAliRashad** [tensorflow/addons#2561]
One of the latest loss functions proposed last month for MultiLabel Classification was in this [paper](https://arxiv.org/pdf/2108.10566v1.pdf), where the authors developed a differentiable confusion matrix to make the F1 Score not only an evaluation matrix but also a loss funciton.

**Comment by old-school-kid** [comment]
Hi @bhack 
Can you please look into the PR #2587

**Comment by old-school-kid** [comment]
Oh oww
So will it considered once the citation threshold is crossed?
